test/e2e/workload_api_test.go

```diff
@@ -301,3 +301,35 @@ func TestDockerUpgradeWorkloadClusterScaleAddRemoveWorkerNodeGroupsAPI(t *testin
 		),
 	)
 }
+
+func TestDockerKubernetes123to124UpgradeFromLatestMinorReleaseAPI(t *testing.T) {
+	release := latestMinorRelease(t)
+	provider := framework.NewDocker(t)
+	managementCluster := framework.NewClusterE2ETest(
+		t,
+		provider,
+		framework.WithClusterFiller(api.WithKubernetesVersion(v1alpha1.Kube123)),
```

Comment:
Pls use `WithClusterConfig` for this

Reply:
I used this for this test because `CreateManagementClusterForVersion` calls `GenerateClusterConfigForVersion` which seems to have some version cleanup logic in that function. Does that matter?

If not, I can simply go with using `WithClusterConfig`  and then use `test.CreateManagementCluster(framework.ExecuteWithEksaRelease(release))`